parser/internal/parservX/parser.go

```diff
@@ -818,6 +831,141 @@ func (p *parser) parse() (ast.Expression, error) {
 	return result, nil
 }
 
+func (p *parser) parseFunctionDefinitions() error {
+	for {
+		tok, lit, _ := p.scanIgnoreWhitespace()
+		if tok == ast.Name && lit == "def" {
```

Comment:
```suggestion
		if tok == ast.Name && isFunctionDefinitionKeyword(lit) {
```

I'm still a fan of initially supporting more variants here so we can get feedback from people:

```go
func isFunctionDefinitionKeyword(s string) {
	switch s {
	case "def", "func", "functions", ???:
		return true
	default:
		return false
	}
}
```

Reply:
yeah, I was going to address this separately.